pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -158,7 +158,7 @@ public Response downloadSegment(
     boolean hasDataAccess;
     try {
       AccessControl accessControl = _accessControlFactory.create();
-      hasDataAccess = accessControl.hasDataAccess(httpHeaders, tableName);
+      hasDataAccess = accessControl.hasAccess(tableName, AccessType.READ, httpHeaders, null);
```

Comment:
@abhioncbr Can we please pass the endpointUrl (/methodName) here and in PinotQueryResource (ideally we should pass it in all places where `hasAccess` is called). 

Internally at Linkedin, we have different restrictions on READ access for downloadSegment and query endpoints.

Reply:
Sure, I'll make the required changes. Thanks